js/tree.js

```diff
@@ -325,32 +332,156 @@
+
+			//trigger callback when all folders have reported opened
+			self.$element.on('loaded.fu.tree', openReported);
+
+			// open all visible folders
+			self.$element.find(".tree-branch:not('.tree-open, .hide')").each(function triggerOpen() {
+				self.openFolder($(this).find('.tree-branch-header'), true);
 			});
+		},
+
+		/**
+		* Disclose all will keep listening for `loaded.fu.tree` and if `$(tree-el).data('keep-disclosing')`
+		* is `true` (defaults to `true`) it will attempt to disclose any new closed folders than were
+		* loaded in during the last disclosure.
+		*/
+		discloseAll: function discloseAll() {
+			var self = this;
+
+			//first time
+			if (typeof self.$element.data('disclosures') === 'undefined') {
+				self.$element.data('disclosures', 0);
+				self.$element.data('keep-disclosing', true);
+			}
+
+			//stop the recursion if they throw the kill switch, or if there isn't anything left to do
+			if (self.$element.data('keep-disclosing') && self.$element.find(".tree-branch:not('.tree-open, .hide')").length !== 0) {
+				/*
+				* give the dev a hook to listen for exceeding disclusure limit `exceededDisclosuresLimit`.
+				* We can't do `disclosures >= upperLimit` because then if they call it again it will
+				* exit immediately unless they "reset the machine". If instead we check to see if
+				* `disclosures` is a multiple of `upperLimit`, it will run exactly `upperLimit` times
+				* while preserving (as an accurate data point for the dev) the number of `disclosures`
+				* that have occurred on this element since page load
+				*/
+				self.$element.data('disclosures', self.$element.data('disclosures') + 1);
+				if (self.options.disclosuresUpperLimit >= 1 && (self.$element.data('disclosures') % self.options.disclosuresUpperLimit) === 0) {
+					/*
+					* This trigger will fire at the beginning of the upper limit recursion layer
+					* (prior to the calling of `discloseVisible()`). So, if you set the
+					* `disclosuresUpperLimit` to 4, this trigger will fire at the _apparent end_
+					* of round 3, just before round 4 occurs. This could theoretically give you time
+					* to prompt the user if they'd like to continue waiting or not before round 4
+					* ends, changing whether the recursion gets killed or not.
+					*/
+					self.$element.trigger('exceededDisclosuresLimit.fu.tree', {
+						tree: self.$element,
+						disclosures: self.$element.data('disclosures')
+					});
+				} else if (self.options.disclosuresUpperLimit >= 1 && self.$element.data('disclosures') > self.options.disclosuresUpperLimit) {
+					/*
+					* Give the dev an out at each execution following the upper limit execution.
+					* If this is not the first time `discloseAll` has been called, this will
+					* likely fire _each round_ because the tree remembers the total disclosures since
+					* page load, _even if you've called `closeAll`. In order to prevent that, you will
+					* need to manually reset `disclosures` if you call `closeAll`:
```

Comment:
"Manually reset `disclousures`"? Really? What's the reason that closeAll doesn't take care of this?